### PR TITLE
Multi Domain Sentiment Analysis Dataset (MDSA)

### PR DESCRIPTION
**Pull request by abhishekkrthakur**

### CODE REVIEW

File: datasets/mdsd/mdsd.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+_ALL = "all"
+
+
+class MDSDatasetConfig(datasets.BuilderConfig):
+    def __init__(self, *args, domains=None, **kwargs):
+        super().__init__(*args, version=datasets.Version(_VERSION, ""), **kwargs)
+        self.domains = domains
+
+
+class MDSDataset(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/mdsd/mdsd.py:43]:
```suggestion
class MDSD(datasets.GeneratorBasedBuilder):
```
must match the dataset script name

You'll need to regenerate the dataset_infos.json after this change

**Review by lhoestq — COMMENTED** [review]
Looking pretty good so far !

Can you also add the tags of the dataset card ? It's the yaml part at the top of the dataset card. You can find more info here : https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card

**Comment by abhishekkrthakur** [comment]
I'll come back to this one in sometime :)

**Comment by lhoestq** [comment]
Also if you would use `xml.etree.ElementTree` to parse the XML it would be awesome, because right now you're using an external dependency `xmltodict `

**Comment by abhishekkrthakur** [comment]
> Also if you would use xml.etree.ElementTree to parse the XML it would be awesome, because right now you're using an external dependency xmltodict

Its pseudo xml so elementtree fails. xmltodict seems to be working quite good for this. do we have examples of pseudo xml datasets?

**Comment by lhoestq** [comment]
for the other pseudo xml the text is parsed manually

**Comment by lhoestq** [comment]
Can you add `xmltodict` to the test dependencies in setup.py please to fix the CI please ?

**Comment by lhoestq** [comment]
Also can you add the dataset card with the tags and run `make style` ?

**Comment by lhoestq** [comment]
Hi :) have you had a chance to fix the test dependency and apply `make style` ?

Feel fee to ping me when it's ready for a review

**Comment by albertvillanova** [comment]
Thanks for your contribution, @abhishekkrthakur. Are you still interested in adding this dataset?

We are removing the dataset scripts from this GitHub repo and moving them to the Hugging Face Hub: https://huggingface.co/datasets

We would suggest you create this dataset there. Please, feel free to tell us if you need some help.

**Pull request closed without merging**